### PR TITLE
Tests for leader rotation

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
Need leader rotation tests that use the new test harness

#### Summary of Changes
Added tests to check
1. Leader Rotation
2. Dropping leader in the middle of rotation and checking that network stays up

Tests are currently ignored as these are failing

### CODE REVIEW

File: src/cluster_tests.rs [hunk 1]
```diff
@@ -1,12 +1,16 @@
-/// Cluster independant integration tests
+/// Cluster independent integration tests
```
Comment (aeyakovenko) [inline src/cluster_tests.rs]: :)

File: tests/local_cluster.rs [hunk 2]
```diff
@@ -38,3 +40,64 @@ fn test_spend_and_verify_all_nodes_3() -> () {
...
+    let local = LocalCluster::new(num_nodes, 10_000, 100);
+
+    // Check that all nodes cycle through leader role
+    cluster_tests::rotate_leader_through_all_nodes(
+        &local.entry_point_info,
+        num_nodes,
+        Duration::from_secs(num_nodes as u64 * 60),
+    );
+
+    // Exit the leader node
```
Comment (aeyakovenko) [inline tests/local_cluster.rs]: Can you add an rpc to exit instead?  I’d like the tests to run against s testnet

**Comment by aeyakovenko** [comment]
Take a look at #3084

**Comment by garious** [comment]
@pgarg66, anything happening with this PR?

**Comment by pgarg66** [comment]
@garious I'll look at it today. Leader rotation had issues, so the tests were ignored. I'll see if they can be un-ignored and have some value.

**Comment by pgarg66** [comment]
Killing this PR for now, as there's no clean way of finding the current leader using local cluster. We can resurrect it once dust settles with beacons (or probably Grandview).

**Pull request closed without merging**